comid/rawint.go

```diff
@@ -0,0 +1,330 @@
+// Copyright 2025 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package comid
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/veraison/corim/encoding"
+	"github.com/veraison/corim/extensions"
+)
+
+// RawInt describes an integer value that can be compared with linear order in
+// the target environment. It follows the type-choice pattern  widely
+// used in CoRIM and implements the extensions.ITypeChoiceValue interface
+// https://ietf-rats-wg.github.io/draft-ietf-rats-corim/draft-ietf-rats-corim.html#name-raw-int
+type RawInt struct {
+	Value extensions.ITypeChoiceValue
```

Comment:
The extensions framework is for types that is for `extensible-type-choice` as per 
https://ietf-rats-wg.github.io/draft-ietf-rats-corim/draft-ietf-rats-corim.html#name-typographical-conventions-f

However, raw-int is a closed type choice, any reason why it is made extensible ?

Reply:
`extensions.ITypeChoiceValue` appears to be the interface that all concrete types in CoRIM implement. Is there a different interface that rawint should implement?

Please see the following comment for extensions.ITypeChoiceValue:
```golang
// ITypeChoiceValue is the interface that is implemented by all concrete type
// choice value types. Specific type choices define their own value interfaces
// that embed this one (and possibly include additional methods).
```

For example, closed types like SVN implements this interface also:
```golang
type SVN struct {
	Value ISVNValue
}

type ISVNValue interface {
	extensions.ITypeChoiceValue
}
```